egs/wsj/s5/utils/lang/compute_sentence_probs_arpa.py

```diff
@@ -0,0 +1,150 @@
+    if sub_string in ngram_dict:
+        return -float(ngram_dict[sub_string][0][1:])
+    else:
+        backoff_substring = " ".join(sub_list[:-1])
+        backoff_weight = 0.0 if (backoff_substring not in ngram_dict or len(ngram_dict[backoff_substring]) < 2) \
+                         else -float(ngram_dict[backoff_substring][1][1:])
+        return compute_sublist_prob(sub_list[1:]) + backoff_weight
+
+def compute_begin_prob(sub_list):
+    logprob = 0
+    for i in range(1, len(sub_list) - 1):
+        logprob += compute_sublist_prob(sub_list[:i + 1])
+    return logprob
+    
+# The probability is computed in this way:
+# p(word_N | word_N-1 ... word_1) = ngram_dict[word_1 ... word_N][0].
+# Here gram_dict is a dictionary stores a tuple corresponding to ngrams.
+# The first element of tuple is probablity and the second is backoff probability (if exists).
+# If the particular ngram (word_1 ... word_N) is not in the dictionary, then
+# p(word_N | word_N-1 ... word_1) = p(word_N | word_(N-1) ... word_2) * backoff_weight(word_(N-1) | word_(N-2) ... word_1) 
+# If the sequence (word_(N-1) ... word_1) is not in the dictionary, then the backoff_weight gets replaced with 1.0
+# More details can be found in https://cmusphinx.github.io/wiki/arpaformat/
+def compute_sentence_prob(sentence, ngram_order):
+    sentence_split = sentence.split()
+    for i in range(len(sentence_split)):
+        if sentence_split[i] not in ngram_dict:
+            sentence_split[i] = "<unk>"
+    sen_length = len(sentence_split)
+
+    if sen_length < ngram_order:
+        return compute_begin_prob(sentence_split)
+    else:
+        logprob = 0
+        begin_sublist = sentence_split[:ngram_order]
+        logprob += compute_begin_prob(begin_sublist)
+
+        for i in range(sen_length - ngram_order + 1):
+            cur_sublist = sentence_split[i : i + ngram_order]
+            logprob += compute_sublist_prob(cur_sublist)
+
+    return logprob
+        
+
+def output_result(text_in_handle, output_file_handle, ngram_order):
+    index = 1
+    lines = text_in_handle.readlines()
+    for line in lines:
+        new_line = "<s> " + line[:-1] + " </s>"
+        logprob = compute_sentence_prob(new_line, ngram_order)
+        output_file_handle.write("{}\n".format(logprob))
+        index += 1
+    text_in_handle.close()
+    output_file_handle.close()
+     
+            
+if __name__ == "__main__":
+    check_args(args)
+    ngram_dict, tot_num = load_model(args.arpa_lm)
+
+    num_valid, max_ngram_order = check_number(args.arpa_lm, tot_num)
+    if not num_valid:
+        sys.exit("get_sentence_prob.py: Wrong loading model.")
+    if args.ngram_order <= 0 or args.ngram_order > max_ngram_order:
+        sys.exit("get_sentence_prob.py: " +   
+            "Invalid ngram_order (either negative or greater than maximum ngram number ({}) allowed").format(max_ngram_order)
+
+    output_result(args.text_in_handle, args.prob_file_handle, args.ngram_order)
```

Comment:
args.text_in_handle is 'str' type，may be text_in filename,but in line 129 :`lines = text_in_handle.readlines()`, this code must be wrong!